### ISSUE TITLE
pre-trained

### ISSUE THREAD

**Issue by zhending111** [mllam/neural-lam#87]
May I ask if you loaded the pre-trained weights of Graphcast during the model training?

**Comment by sadamov** [comment]
Hi @zhending111, No, I don't think any transfer learning experiments with global weights from graph-cast have been carried out by the community.
 
- Technically: GraphCast originally was built on JAX, its weights can be downloaded in .npz from [Google Cloud Bucket](https://console.cloud.google.com/storage/browser/dm_graphcast). The `graph-lam` model type in Neural-LAM (in Torch) follows roughly the same logic as GraphCast to **build** multiscale graphs. But note: the resulting graph is actually quite different from the original one. As Neural-LAM currently works with rectangular graphs in a limited area setting. Still, if you really wanted to, you might be able to load the original weights of GraphCast to initialize the weights of the MLPs in Neural-LAM. One would just have to make sure that these MLPs have the same dimensions as the original ones. I expect this to require a bit of work, but it might be an interesting exercise.
- From a scientific point of view, it's unclear how much a pre-training with a global model would benefit a limited area model (LAM). The LAM usually operates at higher spatial (1-3km) and temporal resolutions (e.g. 1-3h). The physical processes resolved are quite different (e.g. convection resolving), local features like topography become more relevant. In a LAM setting the "global" information about the atmospheric state is expected to come from another datasource (e.g. a global forecast model). This datasource is used as a forcing of the LAM (i.e. used as input to the model but not used in loss calculation during training). 

So overall, we have been thinking about training a high resolution LAM model that is domain agnostic (i.e. can be used anywhere in the world). Maybe finetuning/transfer learning from such a generic LAM would be more interesting in the future than transfer learning from a globel model like graphcast. Such a generic LAM does not exist yet, so we don't have a checkpoint to share. Could I ask what your general usecase / interest to use neuralLAM would be?

**Comment by joeloskarsson** [comment]
Hi, not sure if you refer to some specific model training from one of the paper, if so please specify further. But in general no, we have not used the jax-weights from the full GraphCast model as initialization for any training. For the global models in https://arxiv.org/abs/2406.04759 we also do not use the weights of the smaller GraphCast* model as initialization for the training of any of the other models.

**Comment by zhending111** [comment]
Thank you very much for your detailed response, which provided a thorough answer to my question.

**Comment by zhending111** [comment]
I have one more question. Have you considered how to directly apply Graphcast to generate high-resolution meteorological data for specific regions of interest? Have you made any attempts in this direction?

**Comment by zhending111** [comment]
> Hi, not sure if you refer to some specific model training from one of the paper, if so please specify further. But in general no, we have not used the jax-weights from the full GraphCast model as initialization for any training. For the global models in https://arxiv.org/abs/2406.04759 we also do not use the weights of the smaller GraphCast* model as initialization for the training of any of the other models.

I have one more question. Have you considered how to directly apply Graphcast to generate high-resolution meteorological data for specific regions of interest? Have you made any attempts in this direction?

**Comment by joeloskarsson** [comment]
Not entirely sure what you mean with "generate high-resolution meteorological data" in this context. If you refer to generating new, synthetic data, then no. Although you could probably achieve something like that by removing the conditioning in our probabilistic forecasting models.

If you mean using the original GraphCast model for higher-resolution regional forecasts, then no that is not something I have considered. We have been more focused on learning new models for the region (that share large parts of the GC architecture), that are trained on such high-resolution data.
